drgn_tools/lock.py

```diff
@@ -124,7 +124,7 @@ def scan_mutex_lock(
     stack: bool,
     time: Optional[int] = None,
     pid: Optional[int] = None,
-) -> None:
+) -> List[Tuple[Object, List[Object]]]:
```

Comment:
Instead of tracking and returning the dependencies, maybe just include an optional argument which is the graph. If the argument is present, then the function would add edges to the graph as it encounters them.